src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1164,6 +1164,31 @@ public void updateFeedbackSession(FeedbackSessionAttributes newSession)
 
         fsDb.updateFeedbackSession(newSession);
     }
+
+    /**
+     * Decrease the response rate of feedback session by 1 if the deleted response is the last response from
+     * the particular student in that feedback session; for use after deleting a student response
+     * @param studentEmail
+     * @param sessionName
+     * @param courseId
+     */
+    public void updateSessionResponseRateForDeletedStudentResponse(String studentEmail, String sessionName,
+            String courseId) throws InvalidParametersException, EntityDoesNotExistException {
+        if (!hasRemainingResponsesFromStudent(studentEmail, sessionName, courseId)) {
+            deleteStudentFromRespondentList(studentEmail, sessionName, courseId);
+        }
+    }
```

Comment:
@damithc 

I think the reason this method feels out of place as a public method is because the bigger operation (that this method is a part of) is a cascading operation.

As such, the various cascading method are necessarily in different `*Logic.java`. For example, there is another method for cascading changing `responseId`: `frcLogic.updateFeedbackResponseCommentsForChangingResponseId()`

A solution will be to name the method something like `cascadeDeletedResponseToAdjustSessionResponseRate()`